### ISSUE TITLE
onionshare doesn't work with Python 3.8 (dictionary keys changed during iteration)

### ISSUE THREAD

**Issue by myf** [micahflee/onionshare#1079]
this issue presents to anyone who's currently using archlinux which comes with `python 3.8`. Launching onionshare in the commandline will fail with the following message:
```
$~ onionshare some_file
OnionShare 2.2 | https://onionshare.org/
OnionShare could not connect to Tor in the background:
dictionary keys changed during iteration
```
After further investigation, this bug is related to the python package `stem` which interfaces with tor. `stem-1.7.1` is not compatible with python3.8.
The discussion of this issue is documented under this ticket https://trac.torproject.org/projects/tor/ticket/32552 . To build a working version, you'll have to build `stem` from master instead `stem1.7.1`:
```
% git clone https://git.torproject.org/stem.git
% cd stem
% sudo python setup.py install
# then go back to onionshare
% sudo pip install --upgrade stem
```
We may have to wait for the new release of stem for the fix to be patched via pypi. But before that, this is a work around if you are using archlinux and/or python 3.8

**Comment by micahflee** [comment]
Stem 1.8 is out https://stem.torproject.org/change_log.html#version-1-8

This may fix the problem...

**Comment by micahflee** [comment]
It worked! I'm using fedora 32 with python 3.8 and stem 1.8.0, and it stem successfully works.

**Comment by mig5** [comment]
Fixed in #1098
